src/cueUtilities.js

```diff
@@ -196,139 +224,168 @@ module.exports = function (params, cy, api) {
-				var cyRenderedPosY = cyRenderedPos.y;
-				var factor = (opts.expandCollapseCueSensitivity - 1) / 2;
-
-				if ( (Math.abs(oldMousePos.x - currMousePos.x) < 5 && Math.abs(oldMousePos.y - currMousePos.y) < 5)
-					&& cyRenderedPosX >= expandcollapseRenderedStartX - expandcollapseRenderedRectSize * factor
-					&& cyRenderedPosX <= expandcollapseRenderedEndX + expandcollapseRenderedRectSize * factor
-					&& cyRenderedPosY >= expandcollapseRenderedStartY - expandcollapseRenderedRectSize * factor
-					&& cyRenderedPosY <= expandcollapseRenderedEndY + expandcollapseRenderedRectSize * factor) {
-					if(opts.undoable && !ur)
-						ur = cy.undoRedo({
-							defaultActions: false
-						});
-					if(api.isCollapsible(node))
-						if (opts.undoable){
-							ur.do("collapse", {
-								nodes: node,
-								options: opts
-							});
-						}
-						else
-							api.collapse(node, opts);
-				else if(api.isExpandable(node))
-					if (opts.undoable)
-						ur.do("expand", {
-							nodes: node,
-							options: opts
-						});
-					else
-						api.expand(node, opts);
-					}
-			}
-		});
+        cy.on('mouseover', 'node', data.eMouseOver = function (e) {
+          let node = this;
+          if (isAGroup(node)) {
+            // clear draws if any
+            if ( hoveredGroup && hoveredGroup.id() != node.id() ) {
+
+              refreshCanvasImages();
+            }
+            if (!selectedGroupsContains(node)) {
+              drawExpandCollapseCue(node);
+              hoveredGroup = node;
+            }
+          } else {
+            // needed incase we hover over a regular node inside a group
+            hoveredGroup = null;
+            refreshCanvasImages();
+          }
+        });
+
+        var oldMousePos = null, currMousePos = null;
+        cy.on('mousedown', 'node', data.eMouseDown = function(e){
+          oldMousePos = e.renderedPosition || e.cyRenderedPosition
+        });
+        cy.on('mouseup', 'node', data.eMouseUp = function(e){
+          currMousePos = e.renderedPosition || e.cyRenderedPosition
+        });
+
+        cy.on('grab', 'node', data.eGrab = function (e) {
+          preventDrawing = true;
+        });
+
+        cy.on('free', 'node', data.eFree = function (e) {
+          preventDrawing = false;
+        });
+
+        cy.on('position', 'node', data.ePosition = function () {
+          if (hoveredGroup) {
+            refreshCanvasImages();
+          }
+        });
+
+        var ur;
+        cy.on('select', 'node', data.eSelect = function(evt){
+          if (this.length > cy.nodes(":selected").length)
+            this.unselect();
+
+          if (options().appearOnGroupSelect) {
+            let node = this;
+
+            if (mouseIsHoveringOver(node)) {
+              hoveredGroup = null;
+            }
+            if (isAGroup(node) && !selectedGroupsContains(node)) {
+              selectedGroups.push(node);
+              refreshCanvasImages();
+            }
```

Comment:
I find the above a bit odd. If you set `hoveredGroup` to null, then surely you're going to want to call `refreshCanvasImages()` as well. But I suspect that `mouseIsHoveringOver()` only returns true when the `if` block below is also true. So I think these two `if` blocks can be refined to be a little more coherent.